### PR TITLE
Remove redundant tasks

### PR DESCRIPTION
**Pull request by rhpvorderman**
This saves a lot of time.

Additionally the adapterclipping.wdl file had only one task left after this. So it was removed and the cutadapt task was moved to QC.wdl

Also cutadapt was changed. This depends on https://github.com/biowdl/tasks/pull/87

I also removed the md5sum checking. It should be in our pipelines somewhere. But NOT in QC. This is a very lumc/biowdl specific thing. So we should include it in our multisample pipelines. By removing this, the pipeline only does QC and nothing more. This allows for reuse by others.

I updated the docs as well to reflect the removals.

### CODE REVIEW

**Comment by DavyCats** [comment]
One could argue that md5sum checks are part of QC (making sure your files aren't corrupt), but I believe these checks were initially part of the readgroup sub-workflows, anyway.

File: QC.wdl [hunk 1]
```diff
@@ -2,19 +2,16 @@ version 1.0
...
     input {
-        FastqPair reads
+        File read1
+        File? read2
         String outputDir
         Boolean alwaysRunAdapterClipping = false
-        String sample
-        String library
-        String readgroup
+        Int minimumReadLength = 2 # Choose 2 here to compensate for cutadapt weirdness. I.e. Having empty or non-sensical 1 base reads.
```
Comment (DavyCats) [inline QC.wdl]: 2 base reads are pretty useless as well.

File: QC.wdl [hunk 2]
```diff
@@ -25,97 +22,64 @@ workflow QC {
...
-
     # if no adapters are found, why run cutadapt? Unless cutadapt is used for quality trimming.
     # In which case alwaysRunCutadapt can be set to true by the user.
     Boolean runAdapterClipping = defined(qualityReportRead1.adapters) || defined(qualityReportRead2.adapters) || alwaysRunAdapterClipping
 
     if (runAdapterClipping) {
-        call AC.AdapterClipping as AdapterClipping {
+        if (defined(read2)) {
+                String read2outputPath = outputDir + "/AdapterClipping/cutadapt_" + basename(select_first([read2]))
+           }
```
Comment (DavyCats) [inline QC.wdl]:
```suggestion
        }
```

File: QC.wdl [hunk 3]
```diff
@@ -25,97 +22,64 @@ workflow QC {
...
-            input:
-                file = select_first([reads.R2]),
-                md5 = select_first([reads.R2_md5])
-        }
-    }
-
     call QR.QualityReport as qualityReportRead1 {
         input:
-            read = reads.R1 ,
+            read = read1 ,
```
Comment (DavyCats) [inline QC.wdl]:
```suggestion
            read = read1,
```

File: QC.wdl [hunk 4]
```diff
@@ -25,97 +22,64 @@ workflow QC {
...
     # In which case alwaysRunCutadapt can be set to true by the user.
     Boolean runAdapterClipping = defined(qualityReportRead1.adapters) || defined(qualityReportRead2.adapters) || alwaysRunAdapterClipping
 
     if (runAdapterClipping) {
-        call AC.AdapterClipping as AdapterClipping {
+        if (defined(read2)) {
+                String read2outputPath = outputDir + "/AdapterClipping/cutadapt_" + basename(select_first([read2]))
+           }
+
+        call cutadapt.Cutadapt {
```
Comment (DavyCats) [inline QC.wdl]: According to the style guidelines, there should be an `as` here.

File: QC.wdl [hunk 5]
```diff
@@ -25,97 +22,64 @@ workflow QC {
...
 
     output {
-        FastqPair readsAfterQC = if runAdapterClipping
-            then select_first([AdapterClipping.afterClipping])
-            else reads
+        File qcRead1 = if runAdapterClipping then select_first([Cutadapt.cutRead1]) else read1
+        File? qcRead2 = if runAdapterClipping then select_first([Cutadapt.cutRead1]) else read2
     }
-}
+ }
```
Comment (DavyCats) [inline QC.wdl]:
```suggestion
}
```

File: docs/index.md [hunk 6]
```diff
@@ -8,9 +8,8 @@ latest: true
 This repository contains a collection of [BioWDL](https://github.com/biowdl)
 workflows which can be used for quality control preprocessing and reporting of
 sequencing data. The following workflows are available:
-- AdapterClipping.wdl: Uses cutadapt to perform adapter clipping.
-- QC.wdl: Combines the other workflows in this repository.
-- QualityReport.wdl: Uses a number of tools to produce quality reports.
+- QC.wdl: Cuts reads using cutadapt on the basis of the quality reports produced by `QualityReport.wdl`.
+- QualityReport.wdl: Uses fastqc of tools to produce quality reports.
```
Comment (DavyCats) [inline docs/index.md]:
```suggestion
- QualityReport.wdl: Uses fastqc to produce quality reports.
```

File: QC.wdl (hunk 1 above)
Comment (rhpvorderman) [reply QC.wdl]: I know. But we have to draw a line somewhere. Minimumreadlength = 1 perhaps? (This was the default at first, but may have crashed some tools)

Comment (DavyCats) [reply QC.wdl]: 2 is fine I guess.

**Pull request merged**